### ISSUE TITLE
Still confused about image preprocessing

### ISSUE THREAD

**Issue by shiluooulihs** [ox-vgg/vgg_face2#48]
hi, WeidiXie,
thank you for sharing the pretrained models! it's very useful!
I want to extract embedded features of some face images (in wild) using your pretrained model.  So, I need to know the exact preprocessing .

In my understanding,
1. first, I should use mtcnn to extrace the bbox of face image, 
2. second, extend bbox by a factor 0.3 and crop the face from image( as example images given in the samples/tight_crop);
3. resize the cropped image with shorter side = 256 or 224 (in the readme, it's 256, but in standard_evaluation/pytorch_feature_extractor.py, it's 224)
4. center 224x224 crop from resized image (RGB format)
5.  normalise image by subtracting the mean given in the downloaded model file (the final image fed to model)
Am i right?

**Comment by LordHarsh** [comment]
I want to use this model "resnet50_256_pytorch". Can anyone tell me how can I do data preprocessing? I am a newbie so please help me out. Currently, the thing confusing me the most is how to use mtcnn to get bounding box and how to crop that that picture with that.

**Comment by WeidiXie** [comment]
Hi, check this code: 

https://github.com/ox-vgg/vgg_face2/blob/master/standard_evaluation/pytorch_feature_extractor.py

**Comment by shiluooulihs** [comment]
hi， @WeidiXie，
yes, i checked the code descripted in standard_evaluation/pytorch_feature_extractor.py as mentioned above.
I just want to confirm the steps to extract embedded features of some face images **in wild**.
And in step3, it is ambiguous to resize the cropped image with shorter side = **256 or 224**

**Comment by LordHarsh** [comment]
@WeidiXie  ,
Thank you for the help.
I tried to run this code after cloning this repo in Google Colab, but I am only getting a complete purple picture. Don't know what is happening. This is the link to Colab notebook, please help me out.

**Comment by WeidiXie** [comment]
Hi, @shiluooulihs 

this is the code for reading images, just pass in the path, it will pre-process the image.
https://github.com/ox-vgg/vgg_face2/blob/921df0a400f599d0b1a201fbfbb9117e6d794e0d/standard_evaluation/pytorch_feature_extractor.py#L15
